### PR TITLE
Ocr3config/return active candidate

### PR DESCRIPTION
**Pull request by 0xAustinWang**
https://smartcontract-it.atlassian.net/browse/CCIP-2687

Updates the home_chain_reader to return an 'ActiveAndCandidate' instance, rather than an array. 
This is necessary for us to distinguish "HOW" to operate on changes when managing state transitions. 
Previously we returned an array of two configurations, with the assumption that there would always be one active. This is not the case with keystone, where new configurations are initialized as candidates.

### CODE REVIEW

File: commit/report.go [hunk 1]
```diff
@@ -158,10 +158,13 @@ func (p *Plugin) ShouldTransmitAcceptedReport(
 }
 
 func (p *Plugin) isCandidateInstance(ctx context.Context) (bool, error) {
+	if p.reportingCfg.ConfigDigest == [32]byte{} {
+		return false, nil
+	}
```
Comment (0xAustinWang) [inline commit/report.go:163]: Is this right? Would this node be active if its config digest was nil?

File: execute/plugin.go [hunk 2]
```diff
@@ -325,12 +325,15 @@ func (p *Plugin) ShouldTransmitAcceptedReport(
 }
 
 func (p *Plugin) isGreenInstance(ctx context.Context) (bool, error) {
```
Comment (makramkd) [inline execute/plugin.go:327]: Lets rename this to `isCandidateInstance` like the other file

File: internal/reader/home_chain.go [hunk 3]
```diff
@@ -244,15 +243,10 @@ func (r *homeChainPoller) GetOCRConfigs(
...
-
-	if allConfigs.CandidateConfig.ConfigDigest != [32]byte{} {
-		ocrConfigs = append(ocrConfigs, allConfigs.CandidateConfig)
-	}
-
-	return ocrConfigs, nil
+	return ActiveAndCandidate{
+		ActiveConfig:    allConfigs.ActiveConfig,
+		CandidateConfig: allConfigs.CandidateConfig,
+	}, nil
```
Comment (makramkd) [inline internal/reader/home_chain.go:249]: I think `allConfigs` is already of type `ActiveAndCandidate` right? Can just return directly

```suggestion
	return allConfigs, nil
```

**Review by 0xnogo — APPROVED** [review]

File: commit/report.go [hunk 4]
```diff
@@ -158,10 +158,13 @@ func (p *Plugin) ShouldTransmitAcceptedReport(
 }
 
 func (p *Plugin) isCandidateInstance(ctx context.Context) (bool, error) {
+	if p.reportingCfg.ConfigDigest == [32]byte{} {
+		return false, nil
```
[the remark this answers is not in the document]
Comment (makramkd) [reply commit/report.go:162]: I don't think libocr would launch an OCR instance with a 0x0 config digest.

File: execute/plugin.go [hunk 5]
```diff
@@ -324,13 +324,16 @@ func (p *Plugin) ShouldTransmitAcceptedReport(
 	return true, nil
 }
 
-func (p *Plugin) isGreenInstance(ctx context.Context) (bool, error) {
+func (p *Plugin) isCandidateInstance(ctx context.Context) (bool, error) {
+	if p.reportingCfg.ConfigDigest == [32]byte{} {
+		return false, nil
```
[the remark this answers is not in the document]
Comment (0xAustinWang) [reply execute/plugin.go:329]: adjusted according to: https://chainlink-core.slack.com/archives/C06S76UEBK8/p1730717056415249

File: commit/report.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (0xAustinWang) [reply commit/report.go:162]: adjusted according to: https://chainlink-core.slack.com/archives/C06S76UEBK8/p1730717056415249

File: internal/reader/home_chain.go [hunk 6]
```diff
@@ -220,9 +220,8 @@ func (r *homeChainPoller) GetFChain() (map[cciptypes.ChainSelector]int, error) {
 
 func (r *homeChainPoller) GetOCRConfigs(
 	ctx context.Context, donID uint32, pluginType uint8,
-) ([]OCR3ConfigWithMeta, error) {
+) (ActiveAndCandidate, error) {
 	var (
-		ocrConfigs []OCR3ConfigWithMeta
 		allConfigs ActiveAndCandidate
```
[the remark this answers is not in the document]
Comment (0xAustinWang) [reply internal/reader/home_chain.go:225]: discussed offline, will rename this in the future.

File: commit/plugin_e2e_test.go [hunk 7]
```diff
@@ -721,7 +721,7 @@ func setupNode(params SetupNodeParams, nodeID commontypes.OracleID) nodeSetup {
 	homeChainReader.EXPECT().GetFChain().Return(fChain, nil)
 	homeChainReader.EXPECT().
 		GetOCRConfigs(mock.Anything, params.donID, consts.PluginTypeCommit).
-		Return([]reader.OCR3ConfigWithMeta{{}}, nil).Maybe()
+		Return(reader.ActiveAndCandidate{}, nil).Maybe()
```
[the remark this answers is not in the document]
Comment (0xAustinWang) [reply commit/plugin_e2e_test.go:724]: discussed offline, will rename this in the future.

File: commit/plugin_e2e_test.go [hunk 8]
```diff
@@ -9,6 +9,8 @@ import (
 	"testing"
 	"time"
 
+	ocrTypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
```
Comment (makramkd) [inline commit/plugin_e2e_test.go:12]: Nit: We've been calling this `ocr3types` elsewhere, nice to stay consistent
```suggestion
	ocr3types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
```

File: execute/test_utils.go [hunk 9]
```diff
@@ -14,6 +15,7 @@ import (
 
 	"github.com/smartcontractkit/libocr/commontypes"
 	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	ocr2t "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
```
Comment (makramkd) [inline execute/test_utils.go:18]:
```suggestion
	ocr3types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
```

**Review by makramkd — APPROVED** [review]

File: commit/plugin_e2e_test.go (hunk 8 above)
Comment (0xAustinWang) [reply commit/plugin_e2e_test.go:12]: there's a collision in the file :( 

We already have an import as `"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"`

Comment (makramkd) [reply commit/plugin_e2e_test.go:12]: Maybe just `ocrtypes` then

**Pull request merged**